integration_test/third_party_apps_test.go

```diff
@@ -377,25 +447,53 @@ func TestThirdPartyApps(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	for _, platform := range testConfig.Platforms {
-		platform := platform // https://golang.org/doc/faq#closures_and_goroutines
-		t.Run(platform, func(t *testing.T) {
-			t.Parallel()
+	tests := []test{}
+	platforms := testConfig.Platforms
+	for _, platform := range platforms {
+		apps, err := appsToTest(agentType, platform)
+		if err != nil {
+			t.Fatalf("Error when reading list of apps to test for agentType=%v, platform=%v. err=%v", agentType, platform, err)
+		}
+		if len(apps) == 0 {
+			t.Fatalf("Found no applications when testing agentType=%v, platform=%v", agentType, platform)
+		}
+		for _, app := range apps {
+			tests = append(tests, test{platform, app, ""})
+		}
+	}
 
-			apps, err := appsToTest(agentType, platform)
-			if err != nil {
-				t.Fatalf("Error when reading list of apps to test for agentType=%v, platform=%v. err=%v", agentType, platform, err)
-			}
-			if len(apps) == 0 {
-				t.Fatalf("Found no applications when testing agentType=%v, platform=%v", agentType, platform)
-			}
-			for _, app := range apps {
-				app := app // https://golang.org/doc/faq#closures_and_goroutines
-				t.Run(app, func(t *testing.T) {
-					t.Parallel()
+	// Filter tests
+	mf := modifiedFiles(t)
+	ia := determineImpactedApps(mf)
```

Comment:
personally I would inline these variables that aren't used for anything later:

	determineTestsToSkip(tests, determineImpactedApps(modifiedFiles(t)), testConfig)

This is spelled out for C++ land in tip of the week 161

but this is an optional suggestion.

Reply:
Done.